apis/python/src/tiledbsoma/options/_soma_tiledb_context.py

```diff
@@ -19,16 +19,30 @@
 from .._types import OpenTimestamp
 from .._util import ms_to_datetime, to_timestamp_ms
 
+if TYPE_CHECKING:
```

Comment:
We should not need this as all the type signatures are quoted.  

Also, given that `tiledb` is no longer a dependency, shouldn't this fail in CI (I wonder why it did not fail?)

Reply:
I had assumed if the modules included in `TYPE_CHECKING` are not found, then it treats the quoted types as string literals instead of types.

Reply:
It treats them as string literals for sure, but I don't think it has anything to do with whether the module is found.  TYPE_CHECKING is a constant that mypy sets to true, and is false during actual "use" of the code.

Easiest solution is to probably just do:  `from tiledb import Ctx` and use "Ctx" rather than "tiledb.Ctx".

Sorry this is such a PITA!

Reply:
This ended up being more complicated than expected, but I think I have something that works now.